src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,57 @@
+package duke;
+
+import duke.commands.AddDeadlineCommand;
+import duke.commands.AddEventCommand;
+import duke.commands.AddToDoCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.UnMarkCommand;
+import duke.exceptions.DukeException;
+
+/**
+ * Represents a <code>Parser</code> that parses input into <code>Command</code> to be executed.
+ */
+public class Parser {
+
+    /**
+     * Parses input into a <code>Command</code>.
+     *
+     * @param fullCommand Input taken from <code>Ui</code>.
+     * @return <code>Command</code> to be executed.
+     * @throws DukeException If input is not of a recognizable pattern.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```

Comment:
I really liked how you abstracted each command into its own class 👍. It makes your code easier to read.